test/jdk/java/text/Format/NumberFormat/LenientParseTest.java

```diff
@@ -128,6 +128,27 @@ public void numFmtStrictIntegerOnlyUsed(String toParse, int expectedValue, int e
         dFmt.setParseIntegerOnly(false);
     }
 
+    // 8335668: Parsing with integer only against String with no integer portion
+    // should fail, not return 0. Expected error index should be 0
+    @Test
+    public void integerParseOnlyFractionOnlyTest() {
+        var fmt = NumberFormat.getIntegerInstance();
+        failParse(fmt, localizeText("."), 0);
+        failParse(fmt, localizeText(".0"), 0);
+        failParse(fmt, localizeText(".55"), 0);
+    }
+
+    // 8335668: Parsing with integer only against String with no integer portion
+    // should fail, not return 0. Expected error index should be 0
+    @Test // Non-localized, run once
+    @EnabledIfSystemProperty(named = "user.language", matches = "en")
+    public void compactIntegerParseOnlyFractionOnlyTest() {
+        var fmt = NumberFormat.getIntegerInstance();
```

Comment:
Should it get a compact number instance, instead of integer?

Reply:
Thanks for catching, I would hope the compact test actually tests a compact format; fixed.